fiftyone/core/state.py

```diff
@@ -63,19 +63,29 @@ def __init__(
         )
         super().__init__()
 
-    def serialize(self):
-        dataset = (
+    def serialize(self, reflective=False):
+        """Serializes the state into a dictionary.
+
+        Args:
+            reflective: whether to include reflective attributes when
+                serializing the object. By default, this is False
+        Returns:
+            a JSON dictionary representation of the object
+        """
+        d = super().serialize(reflective=reflective)
```

Comment:
3132ec0a0 seems to have triggered an ETA-related issue in StateDescription.serialize() - specifically, progress bars won't work in other threads. I'm guessing a progress bar is not desirable in this case (when serializing state for the server in a background thread, since it would interfere with Python shells), but I'm not sure where to disable it.
```
Traceback (most recent call last):
  File ".../fiftyone/server/main.py", line 105, in wrapper
    return func(self, *args, **kwargs)
  File ".../fiftyone/server/main.py", line 188, in on_update
    data["data"]
  File ".../fiftyone/core/state.py", line 75, in serialize
    d = super().serialize(reflective=reflective)
  File ".../eta/eta/core/serial.py", line 347, in serialize
    d[a] = _recurse(getattr(self, a), reflective)
  File ".../eta/eta/core/serial.py", line 493, in _recurse
    return v.to_dict()
  File ".../fiftyone/core/view.py", line 270, in to_dict
    d = super().to_dict(rel_dir=rel_dir)
  File ".../fiftyone/core/collections.py", line 1169, in to_dict
    with fou.ProgressBar() as pb:
  File ".../fiftyone/core/utils.py", line 378, in __init__
    super().__init__(*args, iters_str="samples", quiet=quiet, **kwargs)
  File ".../eta/eta/core/utils.py", line 1091, in __init__
    signal.signal(signal.SIGWINCH, self._update_max_width)
  File "/usr/lib/python3.6/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread
```

Reply:
Turns out that the progress bar is actually being printed to the server's stdout, not to the python shell.
I'll look into a way to disable progress bars from here temporarily.

Reply:
Possible fix in https://github.com/voxel51/fiftyone/pull/559/commits/ba8fd2bdefe24d5658e816b6d81985c3759de01c and https://github.com/voxel51/eta/pull/481